### PR TITLE
Add single cell processes

### PR DESCRIPTION
**Pull request by jvrakor**
@jkokosar 

## Checklist

* [x] Make the processes and tests work
* [x] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [x] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [x] All inputs are used in process.
* [x] All output get saved.

### CODE REVIEW

**Comment by mstajdohar** [comment]
cc @JakaKokosar

**Comment by mstajdohar** [comment]
@JakaKokosar It would be great if @jvrakor's work would be compatible with yours. Please review this PR and sync with @jvrakor on Slack.

File: resolwe_bio/processes/import_data/seq_reads_10x.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ListField,
+    Process,
+    SchedulingClass,
+)
+
+
+class ImportScRNA10x(Process):
+    """Import 10x scRNA reads in FASTQ format."""
+    slug = 'upload-sc-10x'
+    name = 'Uload scRNA 10x fastqs'
```
Comment (jkokosar) [inline resolwe_bio/processes/import_data/seq_reads_10x.py]: Uload -> Upload

File: resolwe_bio/processes/import_data/seq_reads_10x.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        }
+    }
+    data_name = '{{ reads.0.file|default("?") }}'
+
+    class Input:
+        """Input fields to process ImportScRNA10x."""
+        barcodes = ListField(
+            FileField(
+                required=True,
```
Comment (jkokosar) [inline resolwe_bio/processes/import_data/seq_reads_10x.py]: `required=True` is the default, so this is redundant. The same applies bellow.

File: resolwe_bio/processes/scseq/cellranger.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+class CellRangerMkref(Process):
+    """Cell ranger mkref."""
+
+    slug = 'cellranger-mkref'
+    name = 'Cell Ranger Mkref'
+    process_type = 'data:genomeindex:10x'
+    version = '1.0.0'
+    category = 'scRNA-Seq'
+    sheduling_class = SchedulingClass.BATCH
+    entity = {
```
Comment (jkokosar) [inline resolwe_bio/processes/scseq/cellranger.py]: Genome index should not be of entity type Sample

File: resolwe_bio/processes/scseq/cellranger.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        bai_path = os.path.join(output_dir, 'possorted_genome_bam.bam.bai')
+        os.symlink(bam_path, bam_name)
+        os.symlink(bai_path, bai_name)
+        process_inputs = {
+            'src': bam_name,
+            'src2': bai_name,
+            'species': inputs.genome_index.species,
+            'build': inputs.genome_index.build,
+        }
+        self.run_process('upload-bam-indexed', process_inputs)
```
Comment (jkokosar) [inline resolwe_bio/processes/scseq/cellranger.py]: This will likely create a new, separate instance of Sample. Is this as intended?

Comment (jvrakor) [reply resolwe_bio/processes/scseq/cellranger.py]: Good point. That is not intended. Any workaround without writing a new process?

Comment (jkokosar) [reply resolwe_bio/processes/scseq/cellranger.py]: Not that I can think of. We used redundant processes like [this](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/import_data/expressions.yml#L211) before to solve the same issue.

File: resolwe_bio/processes/scseq/cellranger.py [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+        os.symlink(bam_path, bam_name)
+        os.symlink(bai_path, bai_name)
+        process_inputs = {
+            'src': bam_name,
+            'src2': bai_name,
+            'reads': inputs.reads.id,
+            'species': inputs.genome_index.species,
+            'build': inputs.genome_index.build,
+        }
+        self.run_process('upload-bam-scseq-indexed', process_inputs)
```
Comment (jvrakor) [inline resolwe_bio/processes/scseq/cellranger.py]: The process does not get spawned and I do not know why. It works if I run it by itself.

**Pull request merged**